pkgpanda/http/__init__.py

```diff
@@ -0,0 +1,229 @@
+"""Pkgpanda HTTP API"""
+
+import http.client
+import logging
+import os
+import sys
+
+from flask import Flask, current_app, jsonify, request
+
+from pkgpanda import Install, Repository, actions
+from pkgpanda.exceptions import (PackageConflict, PackageError,
+                                 PackageNotFound, ValidationError)
+
+
+empty_response = ('', http.client.NO_CONTENT)
+
+
+def package_listing_response(package_ids):
+    return jsonify(sorted(package_ids))
+
+
+def package_response(package_id, repository):
+    try:
+        package = repository.load(package_id)
+    except ValidationError:
+        response = (
+            error_response("Invalid package ID: {}".format(package_id)),
+            http.client.NOT_FOUND,
+        )
+    except PackageNotFound:
+        response = (
+            error_response('Package {} not found.'.format(package_id)),
+            http.client.NOT_FOUND,
+        )
+    except PackageError:
+        error_message = 'Unable to load package {}.'.format(package_id)
+        logging.exception(error_message)
+        response = (
+            error_response(error_message),
+            http.client.INTERNAL_SERVER_ERROR,
+        )
```

Comment:
These errors `ValidationError, PackageNotFound and PackageError` could be made to accept a an argument, package_id, which and the error message could be contained within the class.  

So the exception could be like

```
except (ValidationError, PackageNotFound, PackageError) as e:
      response = (e.msg, e.http_response)
```

The same pattern could be applied to other places where different types of Errors are getting caught. 

Furthermore, I would suggest the `package_response` can return a `namedtuple` so that parts of it can referenced like `response.msg, response.http_code` .

This way, line 176 below would read like `response.http_code != http.client.OK`.